script/scaffold/templates/config_flow_helper/tests/test_config_flow.py

```diff
@@ -0,0 +1,147 @@
+"""Test the NEW_NAME config flow."""
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.NEW_DOMAIN import async_setup_entry
+from homeassistant.components.NEW_DOMAIN.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+from homeassistant.helpers import entity_registry as er
+
+
+@pytest.mark.parametrize("platform", ("sensor",))
+async def test_config_flow(hass: HomeAssistant, platform) -> None:
+    """Test the config flow."""
+    input_sensor_entity_id = "sensor.input"
+    registry = er.async_get(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
+
+    with patch(
+        "homeassistant.components.NEW_DOMAIN.async_setup_entry", wraps=async_setup_entry
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"name": "My NEW_DOMAIN", "entity_id": input_sensor_entity_id},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "My NEW_DOMAIN"
+    assert result["data"] == {}
+    assert result["options"] == {
+        "entity_id": input_sensor_entity_id,
+        "name": "My NEW_DOMAIN",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    config_entry = hass.config_entries.async_entries(DOMAIN)[0]
+    assert config_entry.data == {}
+    assert config_entry.options == {
+        "entity_id": input_sensor_entity_id,
+        "name": "My NEW_DOMAIN",
+    }
+    assert config_entry.title == "My NEW_DOMAIN"
+
+    # Check the entity is registered in the entity registry
+    assert registry.async_get(f"{platform}.my_NEW_DOMAIN") is not None
+
+    # Check the platform is setup correctly
```

Comment:
We normally don't include tests for the state machine or entity registry in a config flow test. Those should move to the platform tests.